#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "enabled or not", True)
gen.add("diff_range_x", double_t, 0, "rang x differance", 0.1, 0, 1)
gen.add("diff_range_y", double_t, 0, "rang y differance", 0.1, 0, 1)
gen.add("diff_range_z", double_t, 0, "rang z differance", 0.6, 0, 1)
gen.add("diff_max_z", double_t, 0, "maximum differance z", 0.2, 0, 1)
gen.add("z_max", double_t, 0, "global maximum height z", 2.3, 0, 10)

exit(gen.generate("dem_costmap_2d", "costmap_2d", "DEMCostmap2D"))
